src/API/Site/Controllers/MerchantCenter/PriceBenchmarksController.php

```diff
@@ -0,0 +1,231 @@
+		$this->register_route(
+			'mc/price-benchmarks',
+			[
+				[
+					'methods'  => TransportMethods::READABLE,
+					'callback' => $this->get_price_benchmarks_callback(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Maps query arguments from the REST request.
+	 *
+	 * @param Request $request REST Request.
+	 * @return array
+	 */
+	protected function prepare_query_arguments( Request $request ): array {
+		$args = wp_parse_args(
+			array_intersect_key(
+				$request->get_query_params(),
+				$this->get_collection_params()
+			),
+			$request->get_default_params()
+		);
+
+		return $args;
+	}
+
+	/**
+	 * Get the callback function for the price benchmarks request.
+	 *
+	 * @return callable
+	 */
+	protected function get_price_benchmarks_callback(): callable {
+		return function ( Request $request ) {
+			try {
+				/** @var MerchantPriceBenchmarks $merchant */
+				$merchant = $this->container->get( MerchantPriceBenchmarks::class );
+
+				$benchmark_data      = $merchant->get_benchmark_data( $this->prepare_query_arguments( $request ) );
+				$price_insights_data = $merchant->get_price_insights( $this->prepare_query_arguments( $request ) );
+
+				// Map the data to the required format.
+				$response_data = $this->map_price_benchmarks_response( $benchmark_data, $price_insights_data );
+
+				return new Response( $response_data );
+			} catch ( Exception $e ) {
+				return $this->response_from_exception( $e );
+			}
+		};
+	}
+
+	/**
+	 * Maps the benchmark and price insights data to the required API response format.
+	 *
+	 * @param array $benchmark_data Raw benchmark data.
+	 * @param array $price_insights_data Raw price insights data.
+	 * @return array Mapped response data.
+	 */
+	protected function map_price_benchmarks_response( array $benchmark_data, array $price_insights_data ): array {
+		$mapped_data = [];
+
+		if ( empty( $benchmark_data['results'] ) || empty( $price_insights_data['results'] ) ) {
+			return $mapped_data;
+		}
+		// Loop through the benchmark data results.
+		foreach ( $benchmark_data['results'] as $benchmark_result ) {
+			$product_view          = $benchmark_result['productView'];
+			$price_competitiveness = $benchmark_result['priceCompetitiveness'];
+
+			// Find the matching price insights data by offer_id.
+			foreach ( $price_insights_data['results'] as $price_insights_result ) {
+				if ( $product_view['id'] !== $price_insights_result['productView']['id'] ) {
+					continue;
+				}
+
+				$price_insights = $price_insights_result['priceCompetitiveness'] ?? [];
+
+				// Calculate the price gap.
+				$regular_price   = (int) $product_view['priceMicros'];
+				$price_on_google = (int) $price_competitiveness['benchmarkPriceMicros'];
+				$price_gap       = $regular_price - $price_on_google;
+
+				// Get the WooCommerce product ID and thumbnail.
+				$product_id = $this->product_helper->get_wc_product_id( $product_view['id'] );
```

Comment:
Great idea to use this helper, but it doesn't seem to be working, which is causing the unit tests to fail. It looks like we should be able to just use the $product_view['offer_id'] value directly here.

Reply:
Per the response of the queries it will not return `offer_id`. I have introduce the util function similar to `ProductHelper::get_wc_product_id` now it working in codebase and unit tests.

[Price Benchmarks](https://developers.google.com/shopping-content/guides/reports/price-competitiveness) (i.e., price competitiveness)
[Price Suggestions](https://developers.google.com/shopping-content/guides/reports/price-insights) (i.e., price insights)

Reply:
Can we go ahead and merge these PR? Once we get data from MC store and if any adjustment needed we can open follow-up PR. WDYT?